rr_viz/src/ross/mission_dashboard/path_navigation_tools.py

```diff
@@ -0,0 +1,151 @@
+
+        # Buttons
+        self.grid_layout = QGridLayout()
+
+        self.start_recording_button = QPushButton('Start Recording')
+        self.start_recording_button.pressed.connect(self.start_recording)
+
+        self.stop_recording_button = QPushButton('Stop Recording')
+        self.stop_recording_button.pressed.connect(self.stop_recording)
+
+        self.enable_recording_buttons(False)
+        self.set_enable_record.connect(self.enable_recording_buttons)
+
+        self.follow_saved_path_button = QPushButton('Follow Saved Path')
+        self.follow_saved_path_button.pressed.connect(self.start_following)
+
+        self.stop_following_button = QPushButton('Stop Following')
+        self.stop_following_button.pressed.connect(self.stop_following)
+
+        self.enable_following_buttons(False)
+        self.set_enable_follow.connect(self.enable_following_buttons)
+
+        self.grid_layout.addWidget(self.start_recording_button, 0, 0)
+        self.grid_layout.addWidget(self.stop_recording_button, 0, 1)
+        self.grid_layout.addWidget(self.follow_saved_path_button, 1, 0)
+        self.grid_layout.addWidget(self.stop_following_button, 1, 1)
+
+        self.v_layout.addLayout(self.grid_layout)
+
+        self.setLayout(self.v_layout)
+        
+        self.record_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.is_record_up, self.set_enable_record, Hz=1./3.)
+        self.record_state_checker.start()
+
+        self.follow_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.is_record_up, self.set_enable_follow, Hz=1./3.)
+        self.follow_state_checker.start()
+
+    def start_recording(self):
+        path_name, ok = QInputDialog.getText(self, "Path file name", "Specify file name to save recorded path:")
+
+        if ok:
+            if path_name != "":
+                self.file_path = string.replace(self.file_path, "path.txt", path_name + ".txt")
+            
+            self.start_recording_action.send_goal(RecordPathGoal(file_path=self.file_path))
+
+    def stop_recording(self):
+        try:
+            trig_resp = self.stop_recording_srv.call(TriggerRequest())
+            if trig_resp.success:
+                print(trig_resp.message)
+            else:
+                print(trig_resp.message)
+                msg = "Failed to call '" + self.stop_mission_srv_name + "' service"
+                print(msg)
+        except:
+            msg = "Service '" + self.stop_mission_srv_name + "' unavailable"
+            rospy.logwarn(msg)
+
+    def start_following(self):
+        file_path_to_search = string.replace(self.file_path, "/path.txt", "")
+
+        try:
+            trig_resp = self.get_path_files_srv.call(TriggerRequest())
+        except Exception as e:
+            rospy.logwarn_throttle(10, "Failed to fetch path files: {}".format(e))
+            return
+
+        if trig_resp.success:
+            path_files = str(trig_resp.message).split(",")
```

Comment:
I think we are doing lots of operations on file path. How about we make a globally accessible library to simplify these operations?

Reply:
this one isn't on file path. It is taking the returned list of files for following. I named the variable badly. I'll fix it